### ISSUE TITLE
Drop events / Ignore events from Windows Service accounts

### ISSUE THREAD

**Issue by perfecthoogie** [wazuh/wazuh-ruleset#459]
Hello i wonder if anyone could assist, i am currently implumenting an elastic stack with wazuh intergration across out company and am running into some issues currently when trying to filter down some of the logs coming in. 

we have quite a few service accounts that run against our servers and desktops that ping out constant events to the elastic stack system, ive looked around and come across a couple of different methods of dropping these events but i havent been able to get any working so far, 

many of the rules ive tried that have been mentioned on previous posts seem to be out dataed and i get an error listed that the if_sid no longer exists or there is no corosponding rule ID. 

when i have managed to get the rule to take and the service to start back up it dosnt seem to take affect. Currently im writting a rule for a spersific account to be ignored but idealy id like to write a rule so that any account starting with svc_ is ignored. 

I am currently adding these rules to the Var/ossec/etc/rules/local_rules.xml file

is this possible currently ? any assitance would be appriciated.  

currently running the latest version of Wazuh and Elastic within our enviroment as well. 

For further information these are the rules ive tried below 

<group name="exception">

    <rule id="100002" level="0">
    <if_sid>20007,20049</if_sid>
    <field name="EventChannel.EventData.TargetUserName">^svc_splunk</field>
    <description>Windows Logon/Logoff Success (ignored)</description>
    <options>no_full_log</options>
    <group>pci_dss_10.2.5,gpg13_7.1,gpg13_7.2,gdpr_IV_32.2,</group>
  </rule>

    <rule id="100002" level="0">
    <if_sid>18107</if_sid>
    <field name="account_name">svc_splunk</field>
    <description>Ignoring authentication_success to login using splunk user</description>
    <group>authentication_success,pci_dss_10.2.5,gpg13_7.1,gpg13_7.2,</group>
  </rule>

    <rule id="100002" level="0">
    <if_sid>18107,18149</if_sid>
    <id>^528$|^538$|^540$|^4624$</id> 
    <user>^svc_splunk</user>
    <description>Windows Logon Success (ignored)</description>
    <group>authentication_success,pci_dss_10.2.5,gpg13_7.1,gpg13_7.2,</group>
  </rule>

Have even tried to use the rule outlined in the MS_auth config and modifiy slighty adding this into the local_rules config to see if that would work

**Comment by Lopuiz** [comment]
Hello

I would like more information on the issue to give you a solution.
Could you do the following checks and give us following information?

1. Send us the configuration that the Wazuh manager has to process the events.

2.  I'd like you to check if the Wazuh manager processes the desired events.
Activate the [logall](https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/global.html#logall) option in the configuration file `/var/ossec/etc/ossec.conf`.  Modify the `<logall>no</logall>` line by `<logall>yes</logall>` and restart Wazuh.
This will create the file `/var/ossec/logs/archive/archive.log` in which all events that are processed by Wazuh will be written.
Once this is done, execute the appropriate actions to create the events and search for them in `archive.log`.
Send us events to create rules that match it.

Best regards,
Eva

**Comment by perfecthoogie** [comment]
Many thanks for getting back to me Eva, 

just to ask when you say in point 1 


Send us the configuration that the Wazuh manager has to process the events. - are you reffering to the ruleset that i have in the local_rules.xml or the on client ossec conf ? 

appologies im fairly new to the product so im not 100% sure.

**Comment by Lopuiz** [comment]
Don't worry.
I refer to the Windows agent's `ossec.conf` file but it's also a good idea to send us the `local_rules.xml` file so I can reproduce your case.
It is not necessary to send the complete `ossec.conf` file, only the configuration block that sends the events to the manager.

The most important thing is that you see that the events are written in the file `archive.log`. These events are processed by `Analysisd`. And the rules have to match event format.

**Comment by perfecthoogie** [comment]
Hey Lopuiz, 

Many thanks for your message ive attached a couple of files with the details you requests please do let me know if you need anything further 


[local_rules_xml.txt](https://github.com/wazuh/wazuh-ruleset/files/3456878/local_rules_xml.txt)
[Osseconf.txt](https://github.com/wazuh/wazuh-ruleset/files/3456879/Osseconf.txt)
[archivelogentries.txt](https://github.com/wazuh/wazuh-ruleset/files/3456880/archivelogentries.txt)

**Comment by Lopuiz** [comment]
Hi @perfecthoogie 

I'm sorry I confused you. I misunderstood your question.

There are two ways [to collect Windows events](https://documentation.wazuh.com/current/user-manual/capabilities/log-data-collection/how-to-collect-wlogs.html): Eventlog and Eventchannel.

The Eventlog format is inherited from ossec, an example of this format can be the following:
>2016 Oct 20 09:42:00 WinEvtLog: Security: AUDIT_SUCCESS(4656): Microsoft-Windows-Security-Auditing: (no user): no domain: snaowpc: A handle to an object was requested.    Subject:   Security ID:  S-1-5-21-466511811-1859800458-3340017162-1001   Account Name:  Pedro   Account Domain:  snaowpc   Logon ID:  0x1274A0BE    Object:   Source Network Address: - Object Server:  Security   Object Type:  File   Object Name:  C:\ossec_test\perro1 - copia (2).txt   Handle ID:  0x1dd0 Resource Attributes: -    Process Information:   Process ID:  0x1024   Process Name:  C:\Windows\explorer.exe    Access Request Information:   Transaction ID:  {00000000-0000-0000-0000-000000000000}   Accesses:  DELETE      READ_CONTROL      WRITE_DAC      SYNCHRONIZE      ReadData (or ListDirectory)      WriteData (or AddFile)      AppendData (or AddSubdirectory or CreatePipeInstance)      ReadEA      WriteEA      ReadAttributes      Write Attributes         Access Reasons:  -   Access Mask:  0x17019F   Privileges Used for Access Check: -   Restricted SID Count: 0

You use the Eventchannel format, these events are received in JSON format:
>{"win":{"system":{"providerName":"Microsoft-Windows-Security-Auditing","providerGuid":"{54849625-5478-4994-A5BA-3E3B0328C30D}","eventID":"4634","version":"0","level":"0","task":"12545","opcode":"0","keywords":"0x8020000000000000","systemTime":"2019-07-31T23:00:01.288486500Z","eventRecordID":"1434083156","processID":"552","threadID":"6944","channel":"Security","computer":"TH-P-DC1.msad.igl","severityValue":"AUDIT_SUCCESS","message":"An account was logged off."},"eventdata":{"targetUserSid":"S-1-5-21-2839260231-1906545324-4194815861-60441","targetUserName":"svc_splunk","targetDomainName":"MSAD","targetLogonId":"0x19a4960a","logonType":"3"}}}

 Eventchannel is also inherited from ossec but has been modified by Wazuh in version 3.8.
These rules are for version 3.7 or lower, and the manager receives events from the new Eventchannel format (version 3.8 or higher).

There are two issues in the following rule:
```xml
<rule id="100002" level="0">
    <if_sid>20007,20049</if_sid>
    <field name="EventChannel.EventData.TargetUserName">^svc_splunk</field>
    <description>Windows Logon/Logoff Success (ignored)</description>
    <options>no_full_log</options>
    <group>pci_dss_10.2.5,gpg13_7.1,gpg13_7.2,gdpr_IV_32.2,</group>
</rule>
```

1. I can't find the rules ids 20049 and 20007 in the new Wazuh versions.
2. The field `EventChannel.EventData.TargetUserName` belongs to the old Eventchannel

I have achieved that your events do not generate alerts with the following rule:
```xml
  <rule id="100003" level="0">
    <if_sid>60106,60137</if_sid>
    <field name="win.eventdata.targetUserName">^svc_</field>
    <description>Windows Logon Success</description>
    <options>no_full_log</options>
    <group>authentication_success,pci_dss_10.2.5,gpg13_7.1,gpg13_7.2,gdpr_IV_32.2,</group>
  </rule>
```
The `field` change is due to the fact that your events are decoded with the following fields:
```
**Phase 2: Completed decoding.
       decoder: 'json'
       win.system.providerName: 'Microsoft-Windows-Security-Auditing'
       win.system.providerGuid: '{54849625-5478-4994-A5BA-3E3B0328C30D}'
       win.system.eventID: '4634'
       win.system.version: '0'
       win.system.level: '0'
       win.system.task: '12545'
       win.system.opcode: '0'
       win.system.keywords: '0x8020000000000000'
       win.system.systemTime: '2019-07-31T23:00:01.288486500Z'
       win.system.eventRecordID: '1434083156'
       win.system.processID: '552'
       win.system.threadID: '6944'
       win.system.channel: 'Security'
       win.system.computer: 'TH-P-DC1.msad.igl'
       win.system.severityValue: 'AUDIT_SUCCESS'
       win.system.message: 'An account was logged off.'
       win.eventdata.targetUserSid: 'S-1-5-21-2839260231-1906545324-4194815861-60441'
       win.eventdata.targetUserName: 'svc_splunk'
       win.eventdata.targetDomainName: 'MSAD'
       win.eventdata.targetLogonId: '0x19a4960a'
       win.eventdata.logonType: '3'
```
And
```
**Phase 2: Completed decoding.
       decoder: 'json'
       win.system.providerName: 'Microsoft-Windows-Security-Auditing'
       win.system.providerGuid: '{54849625-5478-4994-A5BA-3E3B0328C30D}'
       win.system.eventID: '4624'
       win.system.version: '1'
       win.system.level: '0'
       win.system.task: '12544'
       win.system.opcode: '0'
       win.system.keywords: '0x8020000000000000'
       win.system.systemTime: '2019-07-31T23:00:01.271166100Z'
       win.system.eventRecordID: '1434083155'
       win.system.processID: '552'
       win.system.threadID: '1496'
       win.system.channel: 'Security'
       win.system.computer: 'TH-P-DC1.msad.igl'
       win.system.severityValue: 'AUDIT_SUCCESS'
       win.system.message: 'An account was successfully logged on.'
       win.eventdata.subjectUserSid: 'S-1-5-18'
       win.eventdata.subjectUserName: 'TH-P-DC1$'
       win.eventdata.subjectDomainName: 'MSAD'
       win.eventdata.subjectLogonId: '0x3e7'
       win.eventdata.targetUserSid: 'S-1-5-21-2839260231-1906545324-4194815861-60441'
       win.eventdata.targetUserName: 'svc_splunk'
       win.eventdata.targetDomainName: 'MSAD'
       win.eventdata.targetLogonId: '0x19a4960a'
       win.eventdata.logonType: '3'
       win.eventdata.logonProcessName: 'Advapi'
       win.eventdata.authenticationPackageName: 'MICROSOFT_AUTHENTICATION_PACKAGE_V1_0'
       win.eventdata.workstationName: 'TH-P-DC1'
       win.eventdata.logonGuid: '{00000000-0000-0000-0000-000000000000}'
       win.eventdata.keyLength: '0'
       win.eventdata.processId: '0x228'
       win.eventdata.processName: 'C:\Windows\System32\lsass.exe'
       win.eventdata.ipAddress: '10.241.0.59'
       win.eventdata.ipPort: '64335'
       win.eventdata.impersonationLevel: '%%1833'
```

And the change of `if_sid` is due to the fact that the events match rules 60137 and 60106. These rules will be silenced:

```
**Phase 3: Completed filtering (rules).
       Rule id: '60137'
       Level: '3'
       Description: 'Windows User Logoff'
**Alert to be generated.
```

```
**Phase 3: Completed filtering (rules).
       Rule id: '60106'
       Level: '3'
       Description: 'Windows Logon Success'
**Alert to be generated.
```

If you need to mute other rules or fields you just need to modify the values of the tags.

I hope this helps you. If you still have trouble silencing these events, let us know.

Best regards,
Eva

**Comment by perfecthoogie** [comment]
That's amazing thank you Eva will implement and test today.

**Comment by Lopuiz** [comment]
You're welcome.
I'm glad I was helpful.

Best regards,
Eva

**Comment by perfecthoogie** [comment]
Appolgoies for the delay i have tested this now and can confirm working 

many thanks for your assitance

**Comment by Lopuiz** [comment]
you are welcome!

**Comment by metalcated** [comment]
Sorry to post here but I have a similar sitiuation and tried to replicate this but it's not working to ignore specific hostname(s) events. Should I start a new issue or post here?
